### PR TITLE
fix(lsp): prefer local auto-import specifiers

### LINKED ISSUE #10440: lsp: auto-importing should prefer deps.ts
**Is your feature request related to a problem? Please describe.**

When you import something automatically, it will opt for importing it from a url. It would be nice if that auto-import preferred importing it from deps.ts when that same import is exported from deps.ts

**Describe the solution you'd like**

Auto-imports prioritize importing from deps.ts

### PR DESCRIPTION
**Pull request by nayeemrmn**
Fixes #10440.

Give auto-import completion entries a sort-text suffix depending on if the specifier parses as a URL. This will favour relative and bare (likely import-mapped) specifiers.

### CODE REVIEW

**Comment by nayeemrmn** [comment]
cc @lucacasonato I think this fixes what we talked about in #17867.

File: cli/lsp/tsc.rs [hunk 1]
```diff
@@ -2888,6 +2888,11 @@ impl CompletionEntry {
           }
         }
       }
+      if resolve_url(&source).is_err() {
+        sort_text += "_0";
+      } else {
+        sort_text += "_1";
+      }
```
Comment (dsherret) [inline cli/lsp/tsc.rs:2895]: Can you add a comment explaining what this code is doing? Also, shouldn't it be the opposite or I'm probably reading this wrong or don't understand how the sort text works?

**Review by dsherret — COMMENTED** [review]
Almost there. Just needs a comment in the code explaining what it's doing.

File: cli/lsp/tsc.rs (hunk 1 above)
Comment (nayeemrmn) [reply cli/lsp/tsc.rs:2897]: Lesser sort text means lesser position means higher up on the list. Non-urls (`resolve_url().is_err()`) gives it a lesser sort text suffix.

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**